doc: Update Configurator steps (#7340)
projectacrn/acrn-hypervisor: doc/getting-started/getting-started.rst

      cd ~/acrn-work
      git clone https://github.com/projectacrn/acrn-hypervisor.git
      cd acrn-hypervisor

>       git checkout master # TODO determine what to check out

Reviewer: Eventually this will checkout the v3.0 branch

Reviewer 2: We could use latest tag "acrn-2022w18.4-180000p" for now.

Author: Done

      cd ..

>       git clone --depth 1 --branch release_2.7 https://github.com/projectacrn/acrn-kernel.git # TODO determine what to check out

Reviewer: eventually the acrn-kernel branch name would be release_3.0

Reviewer 2: We could use latest stable tag "acrn-2022w18.4-180000p" for now.

Author: Done.

A **launch script** is a shell script that is used to configure and create a
post-launched User VM. Each User VM has its own launch script.

First, you will install dependencies, build the ACRN Configurator Debian package, and install it on your development computer. Then you will use the ACRN Configurator to generate a scenario configuration file and launch script.

#. On the development computer, install the ACRN Configurator build tools:

   .. code-block:: bash

      sudo apt install -y libwebkit2gtk-4.0-dev \
         build-essential \
         curl \
         wget \
         libssl-dev \
         libgtk-3-dev \
         libappindicator3-dev \
         librsvg2-dev

#. Install Node.js (npm included) as follows:

   a. We recommend using nvm to manage your Node.js runtime. It allows you to
      switch versions and update Node.js easily.

      .. code-block:: bash

         curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash

>    #. In the new terminal, install the latest version of Node.js and npm:

Reviewer: What new terminal?
```suggestion
   #. Rerun your ``.bashrc`` initialization script and then install the latest version of Node.js and npm:
```

      .. code-block:: bash

         source ~/.bashrc
         nvm install node --latest-npm
         nvm use node

#. Install and upgrade Yarn:

   .. code-block:: bash

      npm install --global yarn

#. Install rustup, the official installer for Rust:

   .. code-block:: bash

      curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh

>    At this prompt, type ``1`` and press :kbd:`Enter`.

Reviewer: ```suggestion
   When prompted by the Rust installation script, type ``1`` and press :kbd:`Enter`.
```

   .. code-block:: console

      1) Proceed with installation (default)
      2) Customize installation
      3) Cancel installation
      >1

#. Configure the current shell:

   .. code-block:: bash

      source $HOME/.cargo/env

> #. Install ACRN Configurator dependencies:

Reviewer: ```suggestion
#. Install additional ACRN Configurator dependencies:
```

   .. code-block:: bash

      cd ~/acrn-work/acrn-hypervisor/misc/config_tools/configurator

>       python3 -m pip install -r requirements.txt

Reviewer 3: Any particular reason we're not using `pip3` directly here? I.e. `pip install -r requirements.txt`

Author: @gvancuts Not sure, this command came from Eng team. Should I change it to ``sudo pip3 install -r requirements.txt``?

Reviewer 3: It looks like this is in fact the most robust mechanism to use - see also https://github.com/projectacrn/acrn-hypervisor/pull/7358#discussion_r858672890

Let's keep it as is (and perhaps we should update all `pip install ...` commands to this form (`python3 -m pip install...`) throughout our documentation.

      make configurator

#. Install the ACRN Configurator:

   .. code-block:: bash

      sudo apt install -y ~/acrn-work/acrn-hypervisor/build/acrn-configurator_*_amd64.deb

#. Launch the ACRN Configurator:

   .. code-block:: bash

      acrn-configurator

#. On the left-hand side, look for **Start a new configuration**.
   Confirm that the working folder is ``<path to>/acrn-work/MyConfiguration``. Click **Use This Folder**.

   TODO add screenshots to Configurator steps

#. Import your board configuration file as follows:

   a. Under **Import a board configuration file**, click **Browse for file**.

   #. Browse to ``~/acrn-work/my_board.xml`` and click **Open**.

   #. Click **Import Board File**.

   The ACRN Configurator makes a copy of your board file, changes the file
   extension to ``.board.xml``, and saves the file to the working folder.

#. Create a new scenario as follows:

   a. Under **Create new or import an existing scenario**, click **Create
      Scenario**.

   #. In the dialog box, confirm that **Shared (Post-launched VMs only)** is
      selected.

   #. Confirm that one Service VM and one post-launched VM are selected.

   #. Click **Create**.

> #. Generate the scenario configuration file and launch script:

Reviewer 3: ```suggestion
#. Generate the scenario configuration file and launch scripts:
```
? Or is there just one launch script to be generated?

Reviewer: Just one script (one post-launched VM) in this new GSG

   a. Under **Configure settings for scenario and launch scripts**, the
      scenario's configurable items appear. Feel free to look through all the
      available configuration settings. This is where you can change the
      settings to meet your application's particular needs. But for now, you will update only a few settings to make this example work.

   #. Click the **VM1 Post-launched** tab to access the post-launched VM's
      settings.

   #. Confirm that the Basic Parameters tab is selected, and scroll down to
      **Memory size (MB)**. Change the value to ``1024``. For this example, we
      will use Ubuntu 20.04 to boot the post-launched VM. Ubuntu 20.04 needs at
      least 1024 MB to boot.

   #. Scroll down to **Virtio block device**, click **+**, and enter
      ``~/acrn-work/ubuntu-20.04.4-desktop-amd64.iso``. This parameter
      specifies the VM's OS image and its location on the target system. Later
      in this guide, you will save the ISO file to that directory.

   #. Click **Save Scenario And Launch Scripts** to generate the scenario

>       configuration file and launch script.

Reviewer 3: ```suggestion
      configuration file and launch scripts.
```
? Or is there just one launch script to be generated?

      cd ./build
      ls *.deb

>          acrn-my_board-shared-2.7.deb # TODO update file name

Reviewer: Maybe we should change these references now to be ``acrn-my_board-shared-*.deb`` ?  Does that match the file names you're seeing for the hypervisor build output?

Author: Unfortunately, I haven't been able to build successfully, so I'm not sure what the file name is. I filed a ticket in JIRA with the error details.

Reviewer: In that case, let's just go with it as it is and fix later.
